joomla/joomla-cms
# [6.2] Add community info module for admin dashboard

This is a redo of PR #43748 since new features have to go into 5.2-dev.

**The idea for this PR came from a workshop in which we analysed a** [survey](https://docs.google.com/forms/d/e/1FAIpQLSeZewI5HLnuyd2mpHSjnC_2LrqJ-rsPiugNghhjywnGG0NNgw/viewform?fbzx=-7016279858265720891) **in the joomla community of the D-A-CH region.** The survey and the workshop was organized by Oliver Schuldt and @SniperSister. Thank you very much for your effort.

## What is this module going to improve?

One of the points that was mentioned very frequently in the survey is that there is no good entry point for new, interested volunteers. It is difficult and confusing for newcomers to get the necessary information on how to participate in the Joomla! community. This is what we want to solve with the help of this backend dashboard module.
It provides the most important information and links to join and participate in the Joomla community at the most visible place; The dashboard/cpanel of the backend.
The links and information are retrieved from an API endpoint at joomla.org based on the user's current location. This ensures that the user sees the information that is most relevant to him and minimises the barrier to make it as easy as possible to get started in the community.

The module provides the following information:
- Contact and links to your local Joomla community.
- News about Joomla! and its community.
- Upcoming events in your region.
- Motivation to volunteer.

The current location of the user gets evaluated in the following order. If the location couldnt be found the next approach is used:
1. Manually selected location
2. Automatically detect the [current position](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/getCurrentPosition) by the browser
4. Fallback location: London (UK, lat: 51.5000, long: 0.0000)

## Info about API endpoint

**Main entry point**
In order to fetch local community information, the module calls an endpoint (API). The endpoint used by the module is set in the module settings and gets set during installation. In this PR for testing reasons it is set to be [https://test.joomla.spuur.ch/joomla-community-api/links.php](https://test.joomla.spuur.ch/joomla-community-api/links.php).
**This is just a proof-of-concept endpoint set up to test this PR.** It will be exchanged when the PR is in a state ready to be merged. The final endpoint will be hosted at https://www.joomla.org/. It is thought that we could expand the already existing https://community.joomla.org/user-groups.html directory. Such that every Joomla user group or association can enter the information and links there which they want show to the users close to them in this directory. The API will then fetch the data from the directory and provide it as json response to the module.

**News feed**
The news feed is fetched from a source which provides an XML/RSS feed. The source is declared in the response of the main entry point API. This could e.g be a website with a category blog of Joomla articles from which the view format RSS is requested.
(e.g. https://community.joomla.org/blogs?format=feed&type=rss)

**Events feed**
The upcoming events are fetched from a middle ware API endpoint hosted at the same location as the main entry point API for the links. It will parse a calendar from https://community.joomla.org/events.html based on the current location.
(e.g. https://test.joomla.spuur.ch/joomla-community-api/events.php)

The module can also be found in this repository. Here is an installable version of the module if you dont want to build Joomla from source to test this PR:
https://github.com/Elfangor93/mod_community_info

## Summary of Changes
- Added the module "mod_community_info" to the administrator dashboard on the first possible position on the module position "cpanel".

### Testing Instructions
1. Install Joomla from source (or use the [installable module](https://github.com/Elfangor93/mod_community_info))
2. Activate debug in the Joomla global configuration.
3. Visit the administrator dashboard.
4. Allow your browser to get your current location or click on "Choose location" to manually choose your location.
5. Look at the browser console to see the activities of the module when evaluating your location and fetch content.
6. Exchange endpoint in the module parameters to simulate different locations
    - Location Northern Virginia (USA): https://test.joomla.spuur.ch/joomla-community-api/links-usa.php
    - Location Zürich (CH): https://test.joomla.spuur.ch/joomla-community-api/links.php

### Actual result BEFORE applying this Pull Request
No community module in the administrator dashboard.


### Expected result AFTER applying this Pull Request
New module called "Joomla! Community and News" (mod_community_info) community module in the administrator dashboard.


### Link to documentations
Please select:
